calling-apis/intro-to-asynchronous-programming.md

```diff
@@ -0,0 +1,136 @@
+| ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------- | --------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Flow of control          | The order of execution of programming statements.                                                                                                 | -                 | "When we trace code as it runs, we are following the flow of control"                                                                               |
+| Asynchronous programming | A method of programming that intentionally executes actions and processes outside of program flow, so that the program doesn't stop for a result. | Async programming | "Making API calls and loading pages on the web is usually asynchronous, and other code will execute without waiting for the result of an API call." |
+
+## Flow of Control
+
+[**Flow of control**](https://en.wikipedia.org/wiki/Control_flow) in programming refers to the order of programming statements that are executed, and when.
+
+[Synchronous code](https://developer.mozilla.org/en-US/docs/Glossary/synchronous) is code with a specific _flow of control_. Synchronous code always finishes executing an entire line of code before proceeding to the next line of code.
+
+We have been writing synchronous code this whole time!
+
+For example, observe this Python code and this JavaScript code which do the same thing:
+
+```python
+def say_apples():
+    print("apples")
+
+def say_oranges():
+    print("oranges")
+
+say_apples()
+say_oranges()
+```
+
+```js
+const sayApples = () => {
+  console.log("apples");
+};
+
+const sayOranges = () => {
+  console.log("oranges");
+};
+
+sayApples();
+sayOranges();
+```
+
+Both code snippets print out `apples` first, and then `oranges` second, because we call the function that prints `apples` first, and _then_ we call the function that prints `oranges` second. If we need to change the order of how these print, we need to change the order of the lines of code.
+
+### Pizza Metaphor: Synchronous Code
+
+Unlike the food-delivery example from above, synchronous code _executes and finishes_ in a very specific order.
+
+Synchronous code may be more like calling the restaurant and making the order:
+
+1. The restaurant asks what dishes you want
+1. They wait for your answer
+1. Then they proceed to the next question
+
+## Asynchronous Programming
+
+[Asynchronous Programming](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Concepts) is a method of programming that intentionally breaks program flow in order to:
```

Comment:
similar changes as in the vocab
bring the purpose (avoiding the stopping) up from the methodology
the bullets are actually describing how async achieves the goal of not stopping, so modify the lead-in line to reflect this.
```suggestion
[Asynchronous Programming](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Concepts) is a method of programming that intentionally breaks the typical flow of control in order to avoid having to stop and wait for a result.

Asynchronous programming accomplishes this by:
```